CMakeLists.txt

```diff
@@ -1,111 +1,104 @@
-cmake_minimum_required(VERSION 3.14)
+cmake_minimum_required(VERSION 3.14...3.20)
 
 project(
-    example
-    VERSION 0.1.0
-    LANGUAGES CXX
+  example
+  VERSION 0.1.0
+  LANGUAGES CXX
 )
 
+option(BUILD_SHARED_LIBS "default" YES)
+
 # ---- Warning guard ----
 
 # Protect dependents from this project's warnings if the guard isn't disabled
-set(example_warning_guard SYSTEM)
-if(example_INCLUDE_WITHOUT_SYSTEM)
-  set(example_warning_guard "")
-endif()
+## XXX why? CK
+## set(example_warning_guard SYSTEM)
+## if(example_INCLUDE_WITHOUT_SYSTEM)
+##   set(example_warning_guard "")
+## endif()
 
 # ---- Declare library ----
 
 include(GenerateExportHeader)
 
-set(headers include/example/main.h)
+include(GNUInstallDirs)
+set(EXAMPLE_INCLUDE_DIRECTORY "${CMAKE_INSTALL_INCLUDEDIR}")
+
+set(EXAMPLE_headers include/example/main.h)
 
-set(sources source/main.cpp)
+set(EXAMPLE_sources source/main.cpp)
```

Comment:
There is no `EXAMPLE` target, why the prefix? There is only a single target as well, so again, why the prefix?

Reply:
cmake-lint warnings about lower case cmake variable names!